### PR TITLE
feat(escalating-issues): Auto archive issues after 14 days

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
PR for #49662 

## Objective:
After an issue has been in the Unresolved(ongoing) state for 14 days, it should be automatically moved to the Archived state. This should happen in a cronjob that runs every 6 hours.

### CODE REVIEW

File: src/sentry/tasks/auto_archive_issues.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    name="sentry.tasks.auto_archive_issues.run_auto_archive",
+    queue="auto_transition_issue_states",
+    max_retries=3,
+)  # type: ignore
+@monitor(monitor_slug="auto-archive-job-monitor")
+def run_auto_archive() -> None:
+    """
+    Automatically transition issues that are ongoing for 14 days to archived until escalating.
+    """
+    logger.info("Starting task for sentry.tasks.weekly_escalating_forecast.run_escalating_forecast")
```
Comment (jangjodi) [inline src/sentry/tasks/auto_archive_issues.py:43]:
```suggestion
    logger.info("Starting task for sentry.tasks.auto_archive_issues.run_auto_archive")
```

File: src/sentry/tasks/auto_archive_issues.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            Group.objects.filter(
+                status=GroupStatus.UNRESOLVED,
+                substatus=GroupSubStatus.ONGOING,
+                project_id__in=project_ids,
+                last_seen__gte=datetime.now() - timedelta(days=14),
+            ),
+            step=ITERATOR_CHUNK,
+        ),
+        ITERATOR_CHUNK,
+    ):
```
Comment (barkbarkimashark) [inline src/sentry/tasks/auto_archive_issues.py:78]: I think following the pattern we have for auto transitioning regressed issues to ongoing might make sense.

File: src/sentry/tasks/auto_archive_issues.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                GroupHistory.objects.filter(group=group).order_by("-date_added").first()
+            )
+            if not current_group_history:
+                continue
+
+            if not current_group_history.status == GroupHistoryStatus.ONGOING:
+                continue
+
+            if current_group_history.date_added <= fourteen_days_ago:
+                group.update(status=GroupStatus.IGNORED, substatus=GroupSubStatus.UNTIL_ESCALATING)
```
Comment (barkbarkimashark) [inline src/sentry/tasks/auto_archive_issues.py:91]: We probably still wanna follow the pattern where we do a conditional update on the group to ensure we don't introduce race conditions when concurrently updating `Group.status` and `Group.substatus`. 

```
updated = Group.objects.filter().update()
if updated:
  # do auto archive things
```

File: src/sentry/tasks/auto_archive_issues.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+                remove_group_from_inbox(group)
+
+                Activity.objects.create_group_activity(
+                    group,
+                    ActivityType.SET_IGNORED,
```
Comment (barkbarkimashark) [inline src/sentry/tasks/auto_archive_issues.py:97]: I wonder if it makes sense to define a new `ActivityType.AUTO_SET_IGNORED` to differentiate between manual ignored vs auto ignored?

**Review by barkbarkimashark — APPROVED** [review]
I think this is mostly fine. I'm not too sure about the `ITERATOR_CHUNK` and having the task update all the groups for multiple projects in one task attempt - might run into timeouts, but we can run it and see if it runs into any timeouts.

**Pull request merged**